packages/holocron/src/holocronModule.jsx

```diff
@@ -17,27 +17,88 @@ import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { createSelector } from 'reselect';
 import hoistStatics from 'hoist-non-react-statics';
-import { INIT_MODULE_STATE } from './ducks/load';
 
 import {
   LOAD_KEY,
   REDUCER_KEY,
   MODULES_STORE_KEY,
-} from './constants';
+  INIT_MODULE_STATE,
+} from './ducks/constants';
+
+// Execute deprecated load function and provide deprecation message
+export function executeLoad({ load, ...restProps } = { }) {
+  if (load) {
+    console.warn('The \'load\' function in holocron has been deprecated. Please use \'loadModuleData\' instead.');
+    return load(restProps);
+  }
+  return undefined;
+}
+
+// Dispatch loadModuleData inside a thunk if it exists
+export function executeLoadModuleData(loadModuleData, WrappedComponent, props) {
+  const { dispatch, ...restProps } = props;
+  if (dispatch && loadModuleData) {
+    return dispatch(async (_, getState, { fetchClient }) => loadModuleData({
+      store: { dispatch, getState },
+      fetchClient,
+      ownProps: restProps,
+      module: WrappedComponent,
+    }));
+  }
+  return undefined;
+}
+
+export function getName(WrappedComponent, name) {
+  return WrappedComponent && (WrappedComponent.displayName || WrappedComponent.name || name);
+}
 
+export function getDisplayName(name) {
+  return `HolocronModule(${name})`;
+}
+
+export async function executeLoadingFunctions({
```

Comment:
Refactored `initiateLoad` to a utility async function to allow us to achieve 100% test coverage. This is a pure function and much easier to test when uplifted outside of the HoC and React land. This function could use a further refactor to a React Hook when we're ready to refactor `holocronModule` completely.